packages/template-retail-react-app/app/routes.jsx

```diff
@@ -118,9 +129,88 @@ export const routes = [
     }
 ]
 
-export default () => {
+const componentNameMap = {
+    'Home': Home,
+    'Cart': Cart,
+    'ProductDetail': ProductDetail,
+    'ProductList': ProductList,
+    'PageNotFound': PageNotFound
+}
+
+const resourceableComponentsMap = {
+    category: ProductList,
+    product: ProductDetail
+}
+
+export default async (locals) => {
     const config = getConfig()
-    return configureRoutes(routes, config, {
-        ignoredRoutes: ['/callback', '*']
-    })
+    let configuredRoutes = []
+
+    const seoUrlMappingEnabled = true
+
+    console.log('isClient', !isServerSide)
+    if (!isServerSide) {
```

Comment:
This large condition is unsightly, I would recommend that we try to clean this up by maybe using some utility files that are created and hidden from view in the react sdk